api/app/signals/apps/api/v1/views/questions.py

```diff
@@ -0,0 +1,27 @@
+from datapunt_api.rest import DatapuntViewSet, HALPagination
+from django.db.models import Q
+
+from signals.apps.api.v1.serializers import PublicQuestionSerializerDetail
+from signals.apps.signals.models import Question
+
+
+class PublicQuestionViewSet(DatapuntViewSet):
+    queryset = Question.objects.all()
+
+    serializer_class = PublicQuestionSerializerDetail
+    serializer_detail_class = PublicQuestionSerializerDetail
+    pagination_class = HALPagination
+    view_name = 'question-detail'
+
+    def get_queryset(self, *args, **kwargs):
+        if 'slug' in self.kwargs and 'sub_slug' in self.kwargs:
+            childq = Q(category__parent__slug=self.kwargs['slug']) & Q(category__slug=self.kwargs['sub_slug'])
+            parentq = Q(category__parent=None) & Q(category__slug=self.kwargs['slug'])
+            return self.queryset.filter(
+                childq | parentq
+            )
```

Comment:
If I am correct this will result in an `OR` query. So when there are no questions an empty list is returned. Or this wil either return the questions for a sub_category AND/OR return the questions from the parent category (if they exists).

Not sure how this will hold up 🤔  How will this affect the order of the questions (mixed parent and sub category questions)? And also how can a parent category question influence a sub category question?

Reply:
- avoid multiple rest calls 
- we need to sort the question on parent category first, followed by child. (i have  adjusted this. there's only one parent per child. if we do a order by on the parent object first (nulls are placed on top) we get the result we're expecting)